### PR TITLE
Add cURL to view API Page and add a dedicated Guide

### PR DESCRIPTION
**Pull request by abidlabs**
Adds cURL docs 

Closes: https://github.com/gradio-app/gradio/issues/7331
Closes: https://github.com/gradio-app/gradio/issues/6471
Closes: https://github.com/gradio-app/gradio/issues/8237

### CODE REVIEW

File: guides/08_gradio-clients-and-lite/03_querying-gradio-apps-with-curl.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+> {"event_id": $EVENT_ID}   
+```
+
+```bash
+$ curl -N https://abidlabs-en2fr.hf.space/call/predict/$EVENT_ID
+
+> event: complete
+> data: ["Bonjour, mon ami."]
+```
+
```
Comment (aliabd) [inline guides/08_gradio-clients-and-lite/03_querying-gradio-apps-with-curl.md:27]:
```suggestion


```

File: guides/08_gradio-clients-and-lite/03_querying-gradio-apps-with-curl.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+## Step 2: GET the result
+
+Once you have received the `EVENT_ID` corresponding to your prediction, you can stream the results. Gradio stores these results  in a least-recently-used cache in the Gradio app. By default, the cache can store 2,000 results (across all users and endpoints of the app). 
+
+To stream the results for your prediction, make a `GET` request with the following syntax:
+
+```bash
+$ curl -N $URL/call/$API_NAME/$EVENT_ID
+```
+
```
Comment (aliabd) [inline guides/08_gradio-clients-and-lite/03_querying-gradio-apps-with-curl.md:182]:
```suggestion


```

**Review by aliabd — COMMENTED** [review]
Guide is great! just need to fix the new lines between tips and codeblocks

**Comment by abidlabs** [comment]
@aliabd I tested a few demos and the cURL commands worked but I think it would be great to test a variety of other demos just to make sure everything is working

File: js/app/src/api_docs/img/bash.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+</svg>
```
Comment (aliabd) [inline js/app/src/api_docs/img/bash.svg:3]: this logo doesn't show up on dark mode: 

Screen Shot 2024-06-04 at 4 05 05 PM

Comment (aliabd) [reply js/app/src/api_docs/img/bash.svg:3]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="Layer_1" x="0px" y="0px" viewBox="0 0 150 150" style="enable-background:new 0 0 150 150; background-color: #72a824;" xml:space="preserve"><script xmlns=""/>
<style type="text/css">
	.st0{fill:#FFFFFF;}
</style>
<g>
	<path class="st0" d="M118.9,40.3L81.7,18.2c-2.2-1.3-4.7-2-7.2-2s-5,0.7-7.2,2L30.1,40.3c-4.4,2.6-7.2,7.5-7.2,12.8v44.2   c0,5.3,2.7,10.1,7.2,12.8l37.2,22.1c2.2,1.3,4.7,2,7.2,2c2.5,0,5-0.7,7.2-2l37.2-22.1c4.4-2.6,7.2-7.5,7.2-12.8V53   C126.1,47.8,123.4,42.9,118.9,40.3z M90.1,109.3l0.1,3.2c0,0.4-0.2,0.8-0.5,1l-1.9,1.1c-0.3,0.2-0.5,0-0.6-0.4l0-3.1   c-1.6,0.7-3.2,0.8-4.3,0.4c-0.2-0.1-0.3-0.4-0.2-0.7l0.7-2.9c0.1-0.2,0.2-0.5,0.3-0.6c0.1-0.1,0.1-0.1,0.2-0.1   c0.1-0.1,0.2-0.1,0.3,0c1.1,0.4,2.6,0.2,3.9-0.5c1.8-0.9,2.9-2.7,2.9-4.5c0-1.6-0.9-2.3-3-2.3c-2.7,0-5.2-0.5-5.3-4.5   c0-3.3,1.7-6.7,4.4-8.8l0-3.2c0-0.4,0.2-0.8,0.5-1l1.8-1.2c0.3-0.2,0.5,0,0.6,0.4l0,3.2c1.3-0.5,2.5-0.7,3.6-0.4   c0.2,0.1,0.3,0.4,0.2,0.7l-0.7,2.8c-0.1,0.2-0.2,0.4-0.3,0.6c-0.1,0.1-0.1,0.1-0.2,0.1c-0.1,0-0.2,0.1-0.3,0   c-0.5-0.1-1.6-0.4-3.4,0.6c-1.9,1-2.6,2.6-2.5,3.8c0,1.5,0.8,1.9,3.3,1.9c3.4,0.1,4.9,1.6,5,5C94.7,103.4,92.9,107,90.1,109.3z    M109.6,103.9c0,0.3,0,0.6-0.3,0.7l-9.4,5.7c-0.2,0.1-0.4,0-0.4-0.3v-2.4c0-0.3,0.2-0.5,0.4-0.6l9.3-5.5c0.2-0.1,0.4,0,0.4,0.3   V103.9z M116.1,49.6L80.9,71.3c-4.4,2.6-7.6,5.4-7.6,10.7v43.4c0,3.2,1.3,5.2,3.2,5.8c-0.6,0.1-1.3,0.2-2,0.2   c-2.1,0-4.1-0.6-5.9-1.6l-37.2-22.1c-3.6-2.2-5.9-6.2-5.9-10.5V53c0-4.3,2.3-8.4,5.9-10.5l37.2-22.1c1.8-1.1,3.8-1.6,5.9-1.6   s4.1,0.6,5.9,1.6l37.2,22.1c3.1,1.8,5.1,5,5.7,8.5C122.1,48.4,119.3,47.7,116.1,49.6z"/>
</g>
</svg>
```

Comment (aliabd) [reply js/app/src/api_docs/img/bash.svg:3]: My suggestion replaces it with the colored one from https://bashlogo.com/ 

Screen Shot 2024-06-04 at 4 16 07 PM

Screen Shot 2024-06-04 at 4 16 28 PM

File: js/app/src/api_docs/RecordingSnippet.svelte [hunk 4]
```diff
@@ -8,7 +8,7 @@
 	export let dependencies: Dependency[];
 	export let short_root: string;
 	export let root: string;
-	export let current_language: "python" | "javascript";
+	export let current_language: "python" | "javascript" | "bash";
```
Comment (aliabd) [inline js/app/src/api_docs/RecordingSnippet.svelte:11]: did you forget to finish the code for the API recorder for bash? You never constructed the snippet from the calls like you do for python and js

Screen Shot 2024-06-04 at 4 20 14 PM

File: js/app/src/api_docs/ParametersSnippet.svelte [hunk 5]
```diff
@@ -5,7 +5,7 @@
 	export let is_running: boolean;
 	export let endpoint_returns: any;
 	export let js_returns: any;
-	export let current_language: "python" | "javascript";
+	export let current_language: "python" | "javascript" | "bash";
```
Comment (aliabd) [inline js/app/src/api_docs/ParametersSnippet.svelte:8]: there's a problem with some types for js and bash 

python: 
Screen Shot 2024-06-04 at 4 27 48 PM

js: 
Screen Shot 2024-06-04 at 4 27 57 PM

bash: 
Screen Shot 2024-06-04 at 4 27 37 PM

File: js/app/src/api_docs/RecordingSnippet.svelte (hunk 4 above)
Comment (abidlabs) [reply js/app/src/api_docs/RecordingSnippet.svelte:11]: Actually I haven't implemented it yet, will do after launch. Right now, the API recorder button just does not appear if you select bash. Will double check this and open an issue to implement later

Comment (abidlabs) [reply js/app/src/api_docs/RecordingSnippet.svelte:11]: https://github.com/gradio-app/gradio/issues/8457

File: js/app/src/api_docs/ParametersSnippet.svelte (hunk 5 above)
Comment (abidlabs) [reply js/app/src/api_docs/ParametersSnippet.svelte:8]: Looked into this issue and its actually not related to this PR. For now, I've put in a temporary fix that if no type information is available, it prints `any` instead:

image

**Comment by abidlabs** [comment]
Ok @aliabd I believe I've made all of the changes! If you'd like to take another look, go ahead. I'll probably merge in ~45 min or so

File: js/app/src/api_docs/ParametersSnippet.svelte [hunk 6]
```diff
@@ -5,7 +5,8 @@
...
+	console.log(js_returns);
```
Comment (aliabd) [inline js/app/src/api_docs/ParametersSnippet.svelte]: causing test to fail

Comment (aliabd) [reply js/app/src/api_docs/ParametersSnippet.svelte]:
```suggestion
```

**Review by aliabd — APPROVED** [review]
Everything looks good to me!

**Comment by abidlabs** [comment]
Thank you so much @aliabd!

**Pull request merged**